bin/install

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env sh
+set -e
+
+# This script is meant for quick & easy install via:
+#   $ sudo sh -c "$(curl -fsSL https://raw.githubusercontent.com/jderusse/docker-dns-gen/master/bin/install)"
+
+
+C1='\033[0;31m'
+C2='\033[0;33m'
+C3='\033[0;32m'
+C4='\033[0;36m'
+C5='\033[0;35m'
+C0='\033[0m'
+
+if [ "$(id -u)" -ne 0 ]; then
+  echo "${C1}This script must be run by root${C0}"
+  echo "${C2} running ${C3}sudo $0 $@${C0}"
+  sudo $0 "$@"
+  exit 0
+fi
+
+GATEWAY=$(ip -4 addr show docker0 | grep -oP '(?<=inet\s)\d+(\.\d+){3}' || echo "")
+main () {
+  requirements
+  install
+}
+
+requirements () {
+  echo "${C2}Requirements${C0}"
+  requirements_packages
+  requirements_services
+  requirements_config
+  requirements_legacy
+  requirements_network
+}
+
+requirements_packages () {
+  install_if_needed "resolvconf" "resolvconf"
+  echo "${C5} - resolvconf is installed${C0}"
+}
+
+requirements_services () {
+  docker info 2>/dev/null >/dev/null || (echo "${C1} - docker should be installed and usable by root user${C0}" && exit 1)
+  echo "${C5} - ${C3}docker${C5} is installed${C0}"
+
+  docker rm -fv dns-gen > /dev/null 2> /dev/null || true
+  echo "${C5} - container ${C3}dns-gen${C5} stoped${C0}"
+}
+
+requirements_config () {
+  sed -i "/# dns-gen/d" /etc/resolvconf/resolv.conf.d/head
+  sed -i "/nameserver ${GATEWAY}/d" /etc/resolvconf/resolv.conf.d/head
+  echo "${C5} - ${C3}resolv.conf${C5} cleaned${C0}"
+}
+
+requirements_legacy () {
+  previous=$(cat /etc/NetworkManager/dnsmasq.d/01_docker 2>/dev/null|grep "127.0.0.1#54" > /dev/null && echo "1" || echo "")
+  if ! [ -z "${previous}" ]; then
+    if confirm "  Should I remove old dnsmasq.d/01_docker file?"; then
+      rm /etc/NetworkManager/dnsmasq.d/01_docker
+      service NetworkManager restart
+      echo "${C5} - previous config ${C3}/etc/NetworkManager/dnsmasq.d/01_docker${C5} removed ${C0}"
+    fi
+  fi
+
+  resolvconf -u
+}
+
+requirements_network () {
+  ip -4 addr show docker0 2>/dev/null >/dev/null || (echo "${C1} - the docker0 interface does not exists${C0}" && exit 1)
+  echo "${C5} - interface ${C3}docker0${C5} exists${C0}"
+
+  used=$(nc -vz -u ${GATEWAY} 53 2>/dev/null && echo "1" || echo "")
+  if [ -z "${used}" ]; then
+    echo "${C5} - port ${C3}53${C5} is free${C0}"
+  else
+    echo "${C1} - port ${C3}53${C1} already used on ${C3}${GATEWAY}${C0}"
+    exit 1
+  fi
+}
+
+install () {
+  echo "${C2}Install${C0}"
+  install_container
+  install_config
+}
+
+install_container () {
+  docker pull jderusse/dns-gen:2
+  docker run -d --name dns-gen \
+        --restart always \
+        --log-opt "max-size=10m" \
+        --net host \
+        -e GATEWAY=$GATEWAY \
+        --volume /:/host \
```

Comment:
Why do you mount all the FS ???

Reply:
To be able to read the host's `/etc/resolv.conf` file (I added a Q/A section in readme to get a full complet answer to your question)